### ISSUE TITLE
Keep permanently injected

### ISSUE THREAD

**Issue by MariusKlug** [araghon007/X1nput#44]
Hey,

Thank you for this tool, it's quite a change on the elite series 2 controller in dirt rally 2.0. I was wondering if it's possible to keep it injected permanently, though, instead of needing to inject it everytime? It's not such a big deal, but a nice qol...

**Comment by araghon007** [comment]
Hi, there is an issue from a couple years ago (#29) I just never had the time to get around to it. I could possibly either make the app start on Windows startup and automatically inject into games if would detect, or use some sort of DLL replacement that would automatically inject into games without even needing the app

**Comment by MariusKlug** [comment]
Ok thanks for the quick response! So right now it is not easily possible. I tried a solution of adding the DLLs as suggested in an old reddit thread (https://www.reddit.com/r/EASPORTSWRC/comments/byd3pq/how_to_get_controller_triggers_to_sort_of_rumble) but that crashed the game on launch.

**Comment by araghon007** [comment]
Yeah, that was the old version. I changed the way it worked completely because I didn't like the way it worked, and wasn't as customizable, lacked some features. You can still download that version here https://github.com/araghon007/X1nput/releases/tag/v2.0

**Comment by instinctualjealousy** [comment]
> Ok thanks for the quick response! So right now it is not easily possible. I tried a solution of adding the DLLs as suggested in an old reddit thread (https://www.reddit.com/r/EASPORTSWRC/comments/byd3pq/how_to_get_controller_triggers_to_sort_of_rumble) but that crashed the game on launch.

I bodged around until I found a few methods for injecting the latest X1nput without the configurator and persistently. I tested two of them. This won't be a tutorial on how to use any of these tools, but it may be useful to some.

### [DxWrapper](https://github.com/elishacloud/dxwrapper) (tested with TOCA3/[Xidi](https://github.com/samuelgr/Xidi)):
It supports injecting arbitrary DLLs with the `LoadCustomDllPath` variable (this would only be useful for 32-bit programs)

### [Ultimate ASI Loader](https://github.com/ThirteenAG/Ultimate-ASI-Loader) (tested with DxWrapper's integrated UAL on TOCA3/Xidi):
X1nput loads successfully as `X1nput(64).asi`- settings INI loads alongside the file regardless of where your ASI is stored- just keep in mind if your game is 64-bit to utilize the ASI loader for that arch and `X1nput64`

### [Hookshot](https://github.com/samuelgr/Hookshot) (nuclear option, not tested):
Hookshot supports injecting arbitrary DLLs with Hookshot.ini- I believe this should work, but it's a pretty undesirable way of going about it

**I'll need to test with more games, games that use XInput natively, 64-bit etc.**

**Comment by instinctualjealousy** [comment]
It works with Forza Horizon 4. Version 3.1, that is.
Here's what I did:
- Dragged the 64-bit Ultimate ASI Loader `dinput8.dll` into the root game directory
- Renamed `bink2w64.dll` to `bink2w64Hooked.dll`
- Renamed `dinput8.dll` to `bink2w64.dll`
- Threw in `X1nput.ini` and `X1nput64.asi` (renamed from `*.dll`)
- Disabled Steam Input

It shouldn't be any less safe than the 2.0 method, but I do feel filthy injecting it on a modern online game like this.

**Comment by kriomi** [comment]
> 它适用于极限竞速地平线 4。即 3.1 版。这是我所做的：
> 
> * 将 64 位 Ultimate ASI Loader 拖到根游戏目录中`dinput8.dll`
> * 重命名为`bink2w64.dll``bink2w64Hooked.dll`
> * 重命名为`dinput8.dll``bink2w64.dll`
> * 投入和（从`X1nput.ini``X1nput64.asi``*.dll`)
> * 禁用蒸汽输入
> 
> 它应该不比 2.0 方法安全，但我确实觉得将它注入这样的现代网络游戏很肮脏。

where can I download X1nput64.asi？

**Comment by instinctualjealousy** [comment]
> > 它适用于极限竞速地平线 4。即 3.1 版。这是我所做的：
> > 
> > * 将 64 位 Ultimate ASI Loader 拖到根游戏目录中`dinput8.dll`
> > * 重命名为`` bink2w64.dll``bink2w64Hooked.dll ``
> > * 重命名为`` dinput8.dll``bink2w64.dll ``
> > * 投入和（从`` X1nput.ini``X1nput64.asi``*.dll ``)
> > * 禁用蒸汽输入
> > 
> > 它应该不比 2.0 方法安全，但我确实觉得将它注入这样的现代网络游戏很肮脏。
> 
> where can I download X1nput64.asi？

It's possible your Chinese translation may have messed something up here.
Download X1nput from the releases as normal. You then rename the file (in your case, X1nput64.dll) to X1nput64.asi. Not all DLLs like being loaded by UAL in this way. It depends on how the injection code works, but it does work with X1nput.

**Comment by kriomi** [comment]
> > where can I download X1nput64.asi？
> 
> It's possible your Chinese translation may have messed something up here. Download X1nput from the releases as normal. You then rename the file (in your case, X1nput64.dll) to X1nput64.asi. Not all DLLs like being loaded by UAL in this way. It depends on how the injection code works, but it does work with X1nput.

thanks a lot  ,it works.  I downloaded winmm.dll from Ultimate ASI Loader into the root game directory, and rename X1nput.ini to X1nput64.asi.  now I don't gotta  launch X1nputConfigurator.exe.
